internal/jsonrpc/monitor.go

```diff
@@ -57,9 +55,18 @@ func performHealthCheck(
 ) error {
 	duration, err := evmClient.HealthCheckRequest(ctx)
```

Comment:
maybe we can make the `err` returned from `HealthCheckRequest` be a custom error type, and have `ErrDNS`, `ErrConnectionRefused` etc to deal with `error`s rathe than a function that returns a string WDYT?